sympy/logic/tests/test_boolalg.py

```diff
@@ -1345,3 +1341,8 @@ def test_relational_threeterm_simplification_patterns_numerically():
                 assert originalvalue == simplifiedvalue, "Original: {}\nand"\
                     " simplified: {}\ndo not evaluate to the same value for"\
                     "{}".format(pattern[0], simplified, sublist)
+
+
+def test_issue_25451():
+    x = Or(And(a, c), Eq(a, b))
+    assert x != And(a, c)
```

Comment:
Test with `!=` is not ideal because many possible objects would pass the test. A better test could be:
```
assert isinstance(x, Or)
assert set(x.args) == {And(a, c), Eq(a, b)}
```

Reply:
Thank you, I will update it now. I appreciate your input.